src/Tizen.MachineLearning.Nnstreamer/Tizen.MachineLearning.Nnstreamer/Commons.cs

```
@@ -0,0 +1,230 @@
+{
+    /// <summary>
+    /// Enumeration for pipeline state.
+    /// </summary>
+    /// <since_tizen> 5.5 </since_tizen>
+    public enum PipelineState
+    {
+        /// <summary>
+        /// Unknown state. Maybe not constructed?
+        /// </summary>
+        Unknown = 0,
+        /// <summary>
+        /// GST-State "Null"
+        /// </summary>
+        Null = 1,
+        /// <summary>
+        /// GST-State "Ready"
+        /// </summary>
+        Ready = 2,
+        /// <summary>
+        /// GST-State "Paused"
+        /// </summary>
+        Pause = 3,
+        /// <summary>
+        /// GST-State "Playing"
+        /// </summary>
+        Playing = 4,
+    };
+
+    /// <summary>
+    /// Possible data element types of Tensor in NNStreamer.
+    /// </summary>
+    /// <since_tizen> 5.5 </since_tizen>
+    public enum TensorType
+    {
+#pragma warning disable CA1720 // Identifier contains type name
+        /// <summary>
+        /// Integer 32bit
+        /// </summary>
+        Int32 = 0,
+        /// <summary>
+        /// Unsigned integer 32bit
+        /// </summary>
+        UInt32,
+        /// <summary>
+        /// Integer 16bit
+        /// </summary>
+        Int16,
+        /// <summary>
+        /// Unsigned integer 16bit
+        /// </summary>
+        UInt16,
+        /// <summary>
+        /// Integer 8bit
+        /// </summary>
+        Int8,
+        /// <summary>
+        /// Unsigned integer 8bit
+        /// </summary>
+        UInt8,
+        /// <summary>
+        /// Float 64bit
+        /// </summary>
+        Float64,
+        /// <summary>
+        /// Float 32bit
+        /// </summary>
+        Float32,
+        /// <summary>
+        /// Integer 64bit
+        /// </summary>
+        Int64,
+        /// <summary>
+        /// Unsigned integer 64bit
+        /// </summary>
+        UInt64,
+        /// <summary>
+        /// Unknown type
+        /// </summary>
+        Unknown,
```

I think, Unknown is not need, 
First, if this enum is used as input, we do not allow Unknown value. 
Second, if this enum is used as output, we do not return Known, we should throw exception instead of returing Unknown

---

I will.